lms/tasks/email_digests.py

```diff
@@ -44,23 +46,47 @@ def send_instructor_email_digest_tasks(*, batch_size):
     now = datetime.now(timezone.utc)
     updated_before = datetime(year=now.year, month=now.month, day=now.day, hour=5)
     updated_after = updated_before - timedelta(days=1)
-    updated_before = updated_before.isoformat()
-    updated_after = updated_after.isoformat()
 
     with app.request_context() as request:  # pylint:disable=no-member
         with request.tm:
+            candidate_courses = (
+                select(Event.course_id)
+                .join(ApplicationInstance)
+                .where(
+                    # Note here that we are considering earlier launches
+                    # We rather take a few more courses that miss some cases
+                    # where the launch that originated the annotations was made around the cutoff time.
+                    Event.timestamp >= updated_after - timedelta(days=1),
+                    Event.timestamp <= updated_before,
+                    # Only courses that belong to AIs with the feature enabled
+                    ApplicationInstance.settings["hypothesis"][
+                        "instructor_email_digests_enabled"
+                    ].astext
+                    == "true",
+                )
+            ).cte("candidate_courses")
```

Comment:
Ok, so `candidate_courses` is the IDs of all the courses with recent assignment launches from an application instance with the feature enabled